### DISCUSSION TITLE
Versioning

### DISCUSSION THREAD

**Discussion by stefanak-michal** [thingsdb/ThingsDB#367] [Q&A]
Hi. 

With my point of view I have few questions. I'm thinking about using ThingsDB as backend technology (no middle layer, no python or anything). So if I want to develop logic I'll be working directly with local ThingsDB instance (through console or GUI). But in the end I've some docker instance which will contain all my logic. 

So if I want to use git ..how? It doesn't make sense to use git on the whole ThingsDB data folder. I don't need "data" in git repository (with "data" I mean informations). How can I store and share only logic, enum, tasks and types (maybe something else what is needed)? If more people will be colaborating, or for deployment.

**Comment by joente** [comment]
The current workflow is to keep the ThinsDB code in a project within `.ti` files. Using the export() method it is possible to generate the ti code to get the current state. With modules, I think we open the option to link the code to a GitHub repository. (Or GitLab once we include support for this platform). Currently, "modules" always start another process which basically "is" the module. This however, could be changed by removing the requirement in `module.json` for the `main` binary. We could then expose code using the module. For example:

```
// code.ti
test = |this| {
   // this refers to the module
  ...
};
```

And in the `module.json` expose the test, something like this:
```json
{
    "exposes": {
         "file": "code.js"
         "test": "@test"
     }
}
```

In ThingsDB the code can then be simple used:
```
mod.test();   // <module_name>.<func()>
```

This idea can be improved, but something in this direction.

**Reply by stefanak-michal** [reply]
You can't call `export` on `@thingsdb`? How to migrate procedures or tasks defined there?
